kernel/src/scan/mod.rs

```diff
@@ -76,6 +76,15 @@ impl ScanBuilder {
         self
     }
 
+    /// Optionally provide an [`Expression`] to filter rows. See [`ScanBuilder::with_predicate`] for
+    /// details. If `predicate_opt` is `None`, this is a no-op.
+    pub fn with_predicate_opt(self, predicate_opt: Option<Expression>) -> Self {
```

Comment:
do we need to accept an option as parameter here? Or is this "convenience" that the caller does not need to to the check whether their expression may be None?

Reply:
Yep, this is a convenience. It makes code a lot simpler, for example [here](https://github.com/delta-incubator/delta-kernel-rs/pull/231/files#diff-5ffbc8cf8f768b9a25b12f4230501aa83831b31c3f8792053f12ae9adcb23561R414)